lib/utils/meta_util.rb

```diff
@@ -0,0 +1,70 @@
+# traverses the page and site objects to find the most appropriate
+# meta data tags based on a predetermined priority
+
+  module Utils
+    class MetaUtil
+      # Priority:
+      # 1) meta image
+      # 2) page image
+      # 3) page background image
+      # 4) page content for 1st image
+      # 5) site variable's image
+      # Don't love it -- some pretty brittle checks based on the naming conventions of present day
+      # method for getting the url for the <meta name="image" content="{{ meta_image }}"> tag in the head of a SSG html page
+      def self.get_meta_image_url(page_meta_image, page_image, page_bg_image, page_content, site_image)
```

Comment:
If you want, we should pair on applying the splat approach to this method sometime.

Reply:
@tcmacdonald So this is what I had in the generator after our discussion where I dropped the img url search bit and treated all the content the same...

```ruby
      def self.get_meta_image_url(*image_args)
        image_url = image_args.find{|image_under_review| has_value?(image_under_review)}.to_s.strip
        prepend_url_protocol(image_url)
      end
```
But calling ```search_for_first_image_url_in_page_content``` within the filter again is the reason I reverted back to the distinct method arguments.